doc/guides/west/repo-tool.rst

```diff
@@ -224,39 +224,142 @@ manifest` file.
+   west update [-f {always,smart}] [-k] [-r]
+               [-g GROUPS] [--stats] [PROJECT ...]
 
-By default, this command:
+**Which projects are updated:**
 
-#. Parses the manifest file, usually :file:`west.yml`
-#. Clones any project repositories that are not already present locally
-#. Fetches any project revisions which are not already pulled from the
-   remote
-#. Sets each project's :ref:`manifest-rev <west-manifest-rev>` branch to the
-   revision specified for that project in the manifest file
-#. Checks out each ``manifest-rev`` in local working trees, as `detached
-   HEADs <https://git-scm.com/docs/git-checkout#_detached_head>`_
+By default, this command parses the manifest file, usually
+:file:`west.yml`, and updates each project specified there.
+If your manifest uses :ref:`project groups <west-manifest-groups>`, then
+only the active projects are updated.
+
+To operate on a subset of projects only, give ``PROJECT`` argument(s). Each
+``PROJECT`` is either a project name as given in the manifest file, or a
+path that points to the project within the workspace. If you specify
+projects explicitly, they are updated regardless of whether they are active.
 
-To operate on a subset of projects only, specify them using the ``PROJECT``
-positional arguments, which can be either project names as given in the
-manifest file, or paths to the local project clones.
+**Project update procedure:**
 
-To force this command to fetch from project remotes even if the revisions
-appear to be available locally, either use ``--fetch always`` or set the
-``update.fetch`` :ref:`configuration option <west-config>` to ``"always"``.
+For each project that is updated, this command:
+
+#. Initializes a local Git repository for the project in the workspace, if
+   it does not already exist
+#. Inspects the project's ``revision`` field in the manifest, and fetches
+   it from the remote if it is not already available locally
+#. Sets the project's :ref:`manifest-rev <west-manifest-rev>` branch to the
+   commit specified by the revision in the previous step.
+#. Checks out ``manifest-rev`` in the local working copy as a `detached
+   HEADs <https://git-scm.com/docs/git-checkout#_detached_head>`_
+#. If the manifest file specifies a :ref:`submodules
+   <west-manifest-submodules>` key for the project, updates the project's
+   submodules as described below.
+
+To avoid unnecessary fetches, ``west update`` will not fetch project
+``revision`` values which are Git SHAs or tags that are already available
+locally. This is the behavior when the ``-f`` (``--fetch``) option has its
+default value, ``smart``. To force this command to fetch from project
+remotes even if the revisions appear to be available locally, either use
+``-f always`` or set the ``update.fetch`` :ref:`configuration option
+<west-config>` to ``always``. SHAs may be given as unique prefixes as
+long as they are acceptable to Git.
```

Comment:
Can you add something like: "as numerous git servers don't allow fetching SHAs directly, west fetches all branches (and tags?) when the revision looks like a SHA and is not found locally. See issue 1234 for future improvements."

Reply:
I've added a commit documenting the existence and reservation of `refs/west`, but that's all I'd like to document for now, as just about everything related to this is an implementation detail.

Reply:
Interesting but that wasn't my point. My main concern is _performance_ misconceptions. `revision:` fields in the manifest are misleading because they look like they let git fetch only what's required - which is indeed possible but only on some servers so west has to be inefficient and slow.

Some background:
https://opensource.googleblog.com/2018/05/introducing-git-protocol-version-2.html
(Note the main optimization described there is way ahead of this)

Reply:
I think this is a bit of a digression for this text so I would like to put it in a footnote. LMK if that's no good.

Reply:
Footnote or anywhere else is fine, this is "only" about documenting some network and storage aspect. Probably west's main performance aspect though.

BTW this would also indirectly explain why updating a `revision: deadbeef` or `revision: cafe5` branch that is not prefixed with `refs/heads/`, can be much slower than updating `revision: java5`, correct?

Reply:
> Footnote or anywhere else is fine, this is "only" about documenting some network and storage aspect. Probably west's main performance aspect though.

OK, that's done so I'm going to leave this as 'resolved'. Thanks.

> BTW this would also indirectly explain why updating a `revision: deadbeef` or `revision: cafe5` branch that is not prefixed with `refs/heads/`, can be much slower than updating `revision: java5`, correct?

Yep.